tests/run_tests.py

```diff
@@ -225,6 +234,27 @@ def _parse_command_line():
 
 if __name__ == "__main__":
 
+    #
+    # Create our own temporary base storage into which everything
+    # will be created. Having a single top-level folder will make
+    # complete deletion very easy.
+    #
+    timestamp_str = datetime.datetime.now().strftime('%Y%m%d_%H%M%S')
+    current_test_run_subdir = "tk-core-tests-%s-%s" % (timestamp_str, uuid.uuid4())
+    new_base_tempdir = os.path.join(tempfile.gettempdir(), current_test_run_subdir)
+    os.makedirs(new_base_tempdir)
```

Comment:
`tempfile.mkdtemp(prefix="tk-core-tests-%s" % timestamp_str)` is probably enough. It will suffix the temp dir with some random data and create the folder.